Content.Tests/DMProject/Tests/Preprocessor/EmptyAndDirectiveMacros.dm

```diff
@@ -0,0 +1,26 @@
+﻿#define EMPTY
+#define EMPTY_FN(a)
+#define DEFINE #define
+#define DEFINE_WITH_EMPTY EMPTY #define
+#define DEFINE_NESTED(NAME, VALUE) \
+	DEFINE NAME VALUE
+#define DEFINE_NESTED_WITH_STATEMENT(NAME, VALUE) \
+	var/test = 4\
+	DEFINE NAME VALUE
+
+DEFINE A 1
+DEFINE_WITH_EMPTY B 2
+DEFINE_NESTED(C, 3)
+EMPTY DEFINE D 4
+ DEFINE E 5
+DEFINE_NESTED_WITH_STATEMENT(F, 6)
```

Comment:
Using this macro makes BYOND throw a fit. I think it causes it to treat every token after it as an attempt at declaring a var?

Reply:
Yeah, I have no idea what the hell BYOND is doing here. Seems like it might be a manifestation of the previous bug with whitespace before a `DEFINE` macro, since `DEFINE_NESTED` works.

Reply:
Okay, after a bit of coderbus discussion, it turns out the issue here is that BYOND eats newlines in multiline macro definitions, and we don't. So for parity, I guess I should make this also eat newlines, in which case the macro should be defined with `var test = 4;` with the semicolon